### PR TITLE
Adds limits subscription to the Telemetry API

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes VIPEROMCT-295

### Describe your changes:
Adds a subscribeToLimits method to the telemetry API
See related PR here: https://github.com/akhenry/openmct-yamcs/pull/333

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/telemetry/TelemetryAPI.js [hunk 1]
```diff
@@ -489,6 +489,62 @@ export default class TelemetryAPI {
...
+   *          the subscription
+   */
+  subscribeToLimits(domainObject, callback) {
+    if (domainObject.type === 'unknown') {
+      return () => {};
+    }
+
+    const provider = this.#findLimitEvaluator(domainObject);
+
+    if (!this.limitsSubscribeCache) {
```
Comment (jvigliotta) [inline src/api/telemetry/TelemetryAPI.js:513]: Could we just initialize this in the constructor?

**Review by jvigliotta — APPROVED** [review]
This looks good to me! I'll wait on @akhenry before merging since it's API.

File: src/api/telemetry/TelemetryAPI.js [hunk 2]
```diff
@@ -489,6 +489,62 @@ export default class TelemetryAPI {
     }.bind(this);
   }
 
+  /**
+   * Subscribe to limits for a specific domain object.
```
Comment (akhenry) [inline src/api/telemetry/TelemetryAPI.js:493]: Maybe we should specifically say `Subscribe to run-time changes in configured telemetry limits` here. Developers might be confused and think we are subscribing to limit violations, which we're not.

**Review by akhenry — CHANGES_REQUESTED** [review]
Looks great! Just one suggestion on documentation. 

Also we need some tests.

File: src/api/telemetry/TelemetryAPI.js (hunk 2 above)
Comment (shefalijoshi) [reply src/api/telemetry/TelemetryAPI.js]: Done

**Comment by shefalijoshi** [comment]
Added tests here: https://github.com/akhenry/openmct-yamcs/pull/333

**Review by akhenry — APPROVED** [review]

**Pull request merged**